lib/honeydew.ex

```diff
@@ -81,14 +81,14 @@ defmodule Honeydew do
       nil
       # Result comes in at 3 seconds
       iex> Honeydew.yield(job, 5000)
-      :pong
+      {:ok, :pong}
       iex> Honeydew.yield(job, 10_000)
       nil # <- because the message has already arrived and been handled
 
   The only time `yield/2` would ever return the result more than once is if
   the job executes more than once (as Honeydew aims for at-least-once execution).
   """
-  @spec yield(Job.t, timeout) :: result | nil
+  @spec yield(Job.t, timeout) :: {:ok, result} | nil
```

Comment:
seems like a `no_return` is called for here, since it can possibly throw an `ArgumentError`

Reply:
Good call. I'll fix it and add a test for it.

Reply:
man, too bad we can't doctest this thing easily :\